nbqa/save_source.py

```diff
@@ -315,14 +295,18 @@ def _should_ignore_code_cell(
     """
     if not source:
         return True
-    ignore = MAGIC + [i.strip() for i in ignore_cells]
-    return any(source[0].lstrip().startswith(i) for i in ignore)
+    process = MAGIC + [i.strip() for i in process_cells]
+    magic_type = MagicHandler.get_ipython_magic_type(source[0])
+    if magic_type != IPythonMagicType.CELL:
+        return False
+    first_line = source[0].lstrip()
+    return all(first_line.split()[0] != f"%%{i}" for i in process)
```

Comment:
`all(first_line.split()[0] != f"%%{i}" for i in process)` could be simplified as `first_line.split()[0].lstrip("%%") not in process`

Reply:
Thanks! Though note that `.lstrip("%%")` is the same as `.lstrip("%")`. I've changed this slightly as `lstrip` is really unintuitive, and using it resulted in bug #500